lib/logflare_web/controllers/plugs/cf_log_push_parser.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Plug.Parsers.LOGPUSH do
+  @moduledoc """
+  Parse Cloudflare Log Push request bodies.
+  """
+  require Logger
+
+  @behaviour Plug.Parsers
+  import Plug.Conn
+  @gzip_header {"content-encoding", "gzip"}
+
+  def init(_params) do
+  end
+
+  def parse(conn, "text", "plain", _headers, _opts) do
+    conn
+    |> read_body()
+    |> decode()
+  end
+
+  @doc false
+  def parse(conn, _type, _subtype, _headers, _opts) do
+    {:next, conn}
+  end
+
+  def decode({:ok, <<>>, conn}) do
+    {:ok, %{}, conn}
+  end
+
+  def decode({:ok, body, conn}) do
+    body =
+      if @gzip_header in conn.req_headers do
+        body |> :zlib.gunzip() |> String.split("\n", trim: true)
```

Comment:
i don't think the test case covers both gzip and non-gzip.

Reply:
No but the only difference is the gunzip so if effectively tests both.  But I can add a test for it.